docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,294 @@
+  Learning Path, return HTTP 409 Conflict.
+
+
+DELETE
+""""""
+
+**Corresponding User Action:** Unenroll
+
+This will follow the same permission logic as the **POST** request allowing
+both self-unenrollment and staff-unenrollment.
+
+* Marks the *active* ``LearningPathEnrollment`` of the learner as inactive, by
+  setting the ``is_active`` field to ``False`` when the learner has an active
+  enrollment.
+* Return a 404 when there are no active enrollments for the learner.
+
+This method will be gated with a Django settings ``LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT``
+and ``LEARNING_PATHS_ALLOW_STAFF_UNENROLLMENT`` that will default to ``False`` and
+will have to be explicitly enabled.
+
+* When the ``LEARNING_PATHS_ALLOW_STAFF_UNENROLLMENT`` is ``False``, the ``is_active``
+  field will be marked as **read-only**, preventing staff from unenrolling learners.
+* When the ``LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT`` is ``False``, DELETE requests
+  from non-staff members will be returned HTTP 403.
+
+
+5. Fetch Enrollments API
+========================
+
+This API would list all the Learning Path enrollments
+
+* of the user making the request, for a non-staff user
+* of all users, for a staff user
+
++---------------------+-------------------------------------------------------+
+| API Path            | /api/v1/learning-path-enrollment/                     |
++---------------------+-------------------------------------------------------+
+| Methods             | GET                                                   |
++---------------------+-------------------------------------------------------+
+| Permissions Required| LoggedIn                                              |
++---------------------+-------------------------------------------------------+
+
+6. Bulk enrollment API
+======================
+
+In order for staff to bulk enroll users into learning paths, implement the
+following API.
+
++---------------------+-------------------------------------------------------+
+| API Path            | /api/v1/learning-path-enrollment/bulk_enroll/         |
++---------------------+-------------------------------------------------------+
+| Methods             | POST                                                  |
++---------------------+-------------------------------------------------------+
+| Permissions Required| Staff or Admin                                        |
++---------------------+-------------------------------------------------------+
+
+The API will accept the following JSON data.
+
+.. code::
+
+   {
+     "learning_paths": "path-v1:ABC+XYZ+2025_Term1,path-v1:CC+DDD+2025_Term1",
+     "emails": "userA@example.com,new_user@example.com",
+     "reason": "reason for bulk enrollment"
+   }
+
+
+* `learning_paths` - a comma separated list of the Learning Path keys to enroll
+  learner into
+* `emails` - a comma separated list of emails of the learners to enroll
+* `reason` - text explaining the reason for the bulk enrollment
```

Comment:
To unblock and merge this PR, maybe we should add a warning about an incomplete scope to this one, too.